util/webhook/webhook.go

```diff
@@ -132,13 +135,22 @@ func (a *ArgoCDWebhookHandler) startWorkerPool(webhookParallelism int) {
 }
 
 func ParseRevision(ref string) string {
+	if ref == "HEAD" {
+		return ref
+	}
 	refParts := strings.SplitN(ref, "/", 3)
 	return refParts[len(refParts)-1]
 }
 
 // affectedRevisionInfo examines a payload from a webhook event, and extracts the repo web URL,
 // the revision, and whether or not this affected origin/HEAD (the default branch of the repository)
-func affectedRevisionInfo(payloadIf interface{}) (webURLs []string, revision string, change changeInfo, touchedHead bool, changedFiles []string) {
+func affectedRevisionInfo(payloadIf interface{}, optionalHandler *ArgoCDWebhookHandler) (webURLs []string, revision string, change changeInfo, touchedHead bool, changedFiles []string) {
```

Comment:
Why not just call this webhookHandler and use it directly without declaring extra vars?

Reply:
fixed it, please check